### ISSUE TITLE
Error: module config version 2.0 not supported

### ISSUE THREAD

**Issue by Michael0126** [bitcraze/toolbelt#8]
When I run "tb build" or "tb make", error occured  as follow:

difan@difan-ThinkStation-P920:~$ cd aideck-esp-firmware/
difan@difan-ThinkStation-P920:~/aideck-esp-firmware$ tb build
**Error: module config version 2.0 not supported**

How to solve this problem?I can't build or make using toolbelt.
Looking forward to your answers.

**Comment by krichardsson** [comment]
Your toolbelt is probably a bit old, try to run

tb update

**Comment by Michael0126** [comment]
> Your toolbelt is probably a bit old, try to run
> 
> tb update

Thanks for your reply! I have tried to run tb update, and got the result as follow:

Pulling bitcraze/toolbelt:latest
Using default tag: latest
latest: Pulling from bitcraze/toolbelt
Digest: sha256:0ccc117d39073ee7961134989744bce2d55a6dedd7467460f97e8430cc88a47d
Status: Image is up to date for bitcraze/toolbelt:latest
docker.io/bitcraze/toolbelt:latest

But when I run "tb build" or "tb make", still report "**Error: module config version 2.0 not supported**".

**Comment by krichardsson** [comment]
What do you get if you check the version?

tb version

**Comment by Michael0126** [comment]
The version is:
bitcraze/toolbelt - 2020-02-28T09:16:49.915077326Z

**Comment by krichardsson** [comment]
Yeah, that is pretty old. Not sure why it does not pull in the latest version, but try 

`docker pull bitcraze/toolbelt`

and check what version you get after that. The expected version is

`bitcraze/toolbelt - 2022-04-14T06:28:20.72864377Z`

**Comment by Michael0126** [comment]
> 

I have run 'docker pull bitcraze/toolbelt'
The version is still bitcraze/toolbelt - 2020-02-28T09:16:49.915077326Z
Is there any other solution to get the latest version of toolbelt?

**Comment by krichardsson** [comment]
This is just standard docker functionality, nothing magic. I don't know why it does not pull in the latest version.

Try to remove your current image and pull it again

```
docker rmi bitcraze/toolbelt:latest
docker pull bitcraze/toolbelt:latest
```

**Comment by Michael0126** [comment]
> This is just standard docker functionality, nothing magic. I don't know why it does not pull in the latest version.
> 
> Try to remove your current image and pull it again
> 
> ```
> docker rmi bitcraze/toolbelt:latest
> docker pull bitcraze/toolbelt:latest
> ```

I have removed my current image and pulled it again.
Still cannot pull in the latest version.
the lasted version is linked to  bitcraze/toolbelt - 2020-02-28T09:16:49.915077326Z
It is so strange. Is it related to the version of docker? My docker version is 20.10.14

**Comment by Michael0126** [comment]
> > This is just standard docker functionality, nothing magic. I don't know why it does not pull in the latest version.
> > Try to remove your current image and pull it again
> > ```
> > docker rmi bitcraze/toolbelt:latest
> > docker pull bitcraze/toolbelt:latest
> > ```
> 
> I have removed my current image and pulled it again. Still cannot pull in the latest version. the lasted version is linked to bitcraze/toolbelt - 2020-02-28T09:16:49.915077326Z It is so strange. Is it related to the version of docker? My docker version is 20.10.14

Is it possible that the latest tag is not linked to bitcraze/toolbelt - 2022-04-14T06:28:20.72864377Z?

**Comment by krichardsson** [comment]
You can find the images here: https://hub.docker.com/r/bitcraze/toolbelt/tags
It looks good to me :-)

I see two possible reasons to why it does not work:

1. You do not get the latest version when you pull it. You can verify if it actually is the latest version by comparing the digest you get when pulling. This is what I get:
```
$ docker pull bitcraze/toolbelt:latest
latest: Pulling from bitcraze/toolbelt
Digest: sha256:0ad8e7a353c6cc8b754c0c3aaf922095ee4782db899f8ebfa9bfe1743065855c
...
```
2. Your `tb` alias is not running the correct version of the image
Is your alias for the `tb` correct? It should be something like

`alias tb='docker run --rm -it -e "HOST_CW_DIR=${PWD}" -e "CALLING_HOST_NAME=$(hostname)" -e "CALLING_UID"=$UID -e "CALLING_OS"=$(uname) -v ${PWD}:/tb-module -v /var/run/docker.sock:/var/run/docker.sock bitcraze/toolbelt'`

**Comment by Michael0126** [comment]
> You can find the images here: https://hub.docker.com/r/bitcraze/toolbelt/tags It looks good to me :-)
> 
> I see two possible reasons to why it does not work:
> 
> 1. You do not get the latest version when you pull it. You can verify if it actually is the latest version by comparing the digest you get when pulling. This is what I get:
> 
> ```
> $ docker pull bitcraze/toolbelt:latest
> latest: Pulling from bitcraze/toolbelt
> Digest: sha256:0ad8e7a353c6cc8b754c0c3aaf922095ee4782db899f8ebfa9bfe1743065855c
> ...
> ```
> 
> 2. Your `tb` alias is not running the correct version of the image
>    Is your alias for the `tb` correct? It should be something like
> 
> `alias tb='docker run --rm -it -e "HOST_CW_DIR=${PWD}" -e "CALLING_HOST_NAME=$(hostname)" -e "CALLING_UID"=$UID -e "CALLING_OS"=$(uname) -v ${PWD}:/tb-module -v /var/run/docker.sock:/var/run/docker.sock bitcraze/toolbelt'`

How can I use the version 'bitcraze/toolbelt:47'? I see it is the latest version, and I can successfully pull it. But I can't use it.

**Comment by krichardsson** [comment]
You can always change your alias to use `bitcraze/toolbelt:47` instead but that will lock it to V47 and it will not be updated later on. 

There must be something in your docker environment that causes this issue, I just don't know what it might be. You could check for all images of the toolbelt by running `docker images` and remove them all with `docker rmi´.

**Comment by Michael0126** [comment]
> You can always change your alias to use `bitcraze/toolbelt:47` instead but that will lock it to V47 and it will not be updated later on.
> 
> There must be something in your docker environment that causes this issue, I just don't know what it might be. You could check for all images of the toolbelt by running `docker images` and remove them all with `docker rmi´.

Thanks for your patience and suggestion. Can you tell me how to change my alias to use v47. I want to use the toolbelt first.

**Comment by krichardsson** [comment]
> Can you tell me how to change my alias to use v47.

Just edit your alias (probably in ~/.bashrc or ~/.profile) and change the `bitcraze/toolbelt` at the very end to `bitcraze/toolbelt:47`
You have to restart your terminal or source the `~/.bashrc` script for the changes to take effect.

**Comment by Michael0126** [comment]
> > Can you tell me how to change my alias to use v47.
> 
> Just edit your alias (probably in ~/.bashrc or ~/.profile) and change the `bitcraze/toolbelt` at the very end to `bitcraze/toolbelt:47` You have to restart your terminal or source the `~/.bashrc` script for the changes to take effect.

I have successfully pulled the 'bitcraze/toolbelt:47' and edit my alias as follow:
"alias tb='docker run --rm -it -e "HOST_CW_DIR=${PWD}" -e "CALLING_HOST_NAME=$(hostname)" -e "CALLING_UID"=$UID -e "CALLING_OS"=$(uname) -v ${PWD}:/tb-module -v ${HOME}/.ssh:/root/.ssh -v /var/run/docker.sock:/var/run/docker.sock bitcraze/toolbelt:47'"

I have restarted my terminal and source the ~/.bashrc. When I run 'tb', It report as follow:
"Error: Could not find module configuration. Are you sure you are located in a module?"

I have checked the images I have pulled in the /var/lib/docker. My docker images are as follow:
REPOSITORY          TAG       IMAGE ID       CREATED        SIZE
bitcraze/aideck     latest    2e6220ef1f0f   7 days ago     5.15GB
bitcraze/toolbelt   47        30e3c6d313fd   13 days ago    339MB
hello-world         latest    feb5d9fea6a5   7 months ago   13.3kB

**Comment by Michael0126** [comment]
> > Can you tell me how to change my alias to use v47.
> 
> Just edit your alias (probably in ~/.bashrc or ~/.profile) and change the `bitcraze/toolbelt` at the very end to `bitcraze/toolbelt:47` You have to restart your terminal or source the `~/.bashrc` script for the changes to take effect.

Thank you so much for your patience and suggestion. But I still cannot use v47...
Is there any problem with my alias in the ~/.bashrc as follow?
alias tb='docker run --rm -it -e "HOST_CW_DIR=${PWD}" -e "CALLING_HOST_NAME=$(hostname)" -e "CALLING_UID"=$UID -e "CALLING_OS"=$(uname) -v ${PWD}:/tb-module -v ${HOME}/.ssh:/root/.ssh -v /var/run/docker.sock:/var/run/docker.sock bitcraze/toolbelt:47'

**Comment by krichardsson** [comment]
> I have restarted my terminal and source the ~/.bashrc. When I run 'tb', It report as follow:
"Error: Could not find module configuration. Are you sure you are located in a module?"

This error is actually not too bad :-) You get it when the toolbelt is running but it could not understand where it is, which means that it appears to work.

The toolbelt is designed to be executed in the root of a bitcraze repository tree, for instance crazyflie-firmware (it is looking for a file named `module.json`). Just `cd` to the root of a repository and execute `tb` which should give you a list of commands that you can execute in this context.
